### PR TITLE
Adding pendo-syncher deployment config

### PR DESCRIPTION
**Pull request by tahmidefaz**
# Overview

This PR is being created to address [this Jira](https://issues.redhat.com/browse/ESSNTL-623).
PR to add the deployment config for the new `pendo-syncher` cronjob.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Comment by tahmidefaz** [comment]
Plan is to run this every 2 hours to collect as much info as possible for debugging purposes before flipping the switch and enable `PENDO_SYNC_ACTIVE`. After which, we will run this once a day.

File: deployment.yaml [hunk 1]
```diff
@@ -337,6 +337,44 @@ objects:
...
+        env:
+          - name: INVENTORY_LOG_LEVEL
+            value: ${LOG_LEVEL}
+          - name: INVENTORY_DB_SSL_MODE
+            value: ${INVENTORY_DB_SSL_MODE}
+          - name: INVENTORY_DB_SSL_CERT
+            value: ${INVENTORY_DB_SSL_CERT}
+          - name: PROMETHEUS_PUSHGATEWAY
+            value: ${PROMETHEUS_PUSHGATEWAY}
+          - name: PENDO_SYNC_ACTIVE
```
Comment (thearifismail) [inline deployment.yaml:359]: Why PENDO_SYNC_ACTIVE is needed in the presence of "suspend"?

**Review by thearifismail — COMMENTED** [review]
Merging should wait until tests in a cluster.

File: deployment.yaml (hunk 1 above)
Comment (tahmidefaz) [reply deployment.yaml:359]: `PENDO_SYNC_ACTIVE` is different from `suspend`. When set to `false`, it does all the pre-processing work but stops short of actually sending the data up the Pendo Api. I added this purely for debugging purposes. I want to make sure that we pre-process everything right, partial invalid data in Pendo will be painful.

**Comment by tahmidefaz** [comment]
@thearifismail Can you give me direction on how to test this in a cluster?

File: deployment.yaml (hunk 1 above)
Comment (thearifismail) [reply deployment.yaml:359]: got it.

**Comment by thearifismail** [comment]
@tahmidefaz Will do after I verify that my ephemeral namespace is working as expected.

File: deployment.yaml [hunk 2]
```diff
@@ -337,6 +337,44 @@ objects:
           requests:
             cpu: 200m
             memory: 256Mi
+    - name: pendo-syncher
+      schedule: ${PENDO_CRON_SCHEDULE}
+      suspend: ${{PENDO_SYNCHER_SUSPEND}}
```
Comment (kruai) [inline deployment.yaml:342]: Not sure the double curly-braces are needed here, but it also wouldn't break anything afaik

**Review by kruai — DISMISSED** [review]
LGTM!

**Review by thearifismail — DISMISSED** [review]
lgtm

**Comment by tahmidefaz** [comment]
@kruai @thearifismail can you guys take a look again? I had to resolve the merge conflict

**Review by kruai — DISMISSED** [review]
LGTM

**Comment by tahmidefaz** [comment]
Sorry @kruai your approval got dismissed again. I actually need the `Log_Level` for `pendo-syncher` to be in `DEBUG` mode for the initial run. We can change it back later on. :crying_cat_face:

**Review by kruai — APPROVED** [review]
That's fine with me. Still LGTM!

**Comment by tahmidefaz** [comment]
Merging this as it had 2 approvals earlier.

**Pull request merged**